src/index.tsx

```diff
@@ -44,42 +39,77 @@ export function validateFormData<T>(
         : undefined;
     }
   }
-  return schema.validate(validateData, { abortEarly: false }).then(
-    () => {
-      setErrors({});
-      return true;
-    },
-    (err: any) => {
-      setErrors(yupToFormErrors(err));
-      return false;
-    }
-  );
+  return schema.validate(validateData, { abortEarly: false });
+}
+
+export interface FormikValues {
+  [field: string]: any;
+}
+
+// @todo make this limited to keys of values
+export interface FormikErrors {
+  [field: string]: string;
 }
 
+// @todo make this limited to keys of values
+// interfact FormikTouched<Values, Field extends keyof Values> ??
```

Comment:
There is probably a nice way to express this in TS. All we are doing is mapings keys to boolean.